Userland/Libraries/LibWeb/Painting/ButtonPaintable.cpp

```diff
@@ -43,17 +43,29 @@ void ButtonPaintable::paint(PaintContext& context, PaintPhase phase) const
 
     auto const& dom_node = layout_box().dom_node();
     if (is<HTML::HTMLInputElement>(dom_node) && phase == PaintPhase::Foreground) {
-        auto text_rect = context.enclosing_device_rect(absolute_rect());
+        auto button_rect = context.enclosing_device_rect(absolute_rect());
+        auto text_rect = button_rect;
+
+        // Apply CSS text-indent property to text rect
+        auto text_indent = computed_values().text_indent().to_px(layout_box(), CSSPixels());
```

Comment:
The second parameter to `to_px()` is the "percentage basis", the value that any percentages are calculated from. The spec for [`text-indent`](https://drafts.csswg.org/css-text-3/#propdef-text-indent) says:

> [Percentages:](https://www.w3.org/TR/css-values/#percentages) 	refers to block container’s own [inline-axis](https://www.w3.org/TR/css-writing-modes-4/#inline-axis) [inner size](https://www.w3.org/TR/css-sizing-3/#inner-size)

So, we need to calculate that and pass it in here, to be fully correct. You don't have to do that now, but if you don't, please leave a `// FIXME: ...` comment here so someone knows to do it in the future.